### PR TITLE
Scala 2.13 + other libs upgrade

### PR DESCRIPTION
**Pull request by mowczare**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -2,35 +2,33 @@ import ReleaseTransformations._
 
 name := "akka-xlsx"
 
-lazy val AkkaVersion        = "2.5.19"
-lazy val AkkaContribVersion = "0.9"
-lazy val AlpakkaXmlVersion  = "0.20"
+lazy val AkkaVersion        = "2.5.31"
```
Comment (ddworak) [inline build.sbt:5]: I'm curious about the reasoning:
- not to go `2.5.32` if performing only bugfix upgrade
- not to go `2.6.10`

File: build.sbt [hunk 2]
```diff
@@ -2,35 +2,33 @@ import ReleaseTransformations._
...
+lazy val AkkaContribVersion = "0.10"
```
Comment (ddworak) [inline build.sbt:6]: Again, there's 0.11 available

File: build.sbt [hunk 3]
```diff
@@ -2,35 +2,33 @@ import ReleaseTransformations._
...
 lazy val commonSettings = Seq(
   updateOptions := updateOptions.value.withGigahorse(false),
-  scalaVersion := "2.12.8",
+  scalaVersion := "2.13.4",
   organization := "de.envisia.akka",
-  scalacOptions in (Compile, doc) ++= Seq(
-    "-target:jvm-1.8",
-    "-deprecation",
+  scalacOptions ++= Seq(
+    "-target:jvm-1.11",
```
Comment (ddworak) [inline build.sbt:17]: Shouldn't be necessary.

File: build.sbt [hunk 4]
```diff
@@ -2,35 +2,33 @@ import ReleaseTransformations._
...
-lazy val AkkaContribVersion = "0.9"
-lazy val AlpakkaXmlVersion  = "0.20"
+lazy val AkkaVersion        = "2.5.31"
+lazy val AkkaContribVersion = "0.10"
+lazy val AlpakkaXmlVersion  = "2.0.2"
 
-lazy val ScalatestVersion  = "3.0.5"
-lazy val ScalacheckVersion = "1.14.0"
+lazy val ScalatestVersion  = "3.0.9"
+lazy val ScalacheckVersion = "1.15.2"
```
Comment (ddworak) [inline build.sbt:10]: Again, both updated partially.

File: project/plugins.sbt [hunk 5]
```diff
@@ -4,5 +4,5 @@ logLevel := Level.Warn
 updateOptions := updateOptions.value.withGigahorse(false)
 
 // Publishing
-addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.1.0")
-addSbtPlugin("com.github.gseitz" % "sbt-release" % "1.0.7")
\ No newline at end of file
+addSbtPlugin("com.jsuereth" % "sbt-pgp" % "2.0.2")
+addSbtPlugin("com.github.gseitz" % "sbt-release" % "1.0.13")
```
Comment (ddworak) [inline project/plugins.sbt:8]: Neither of these should actually be needed in our case. We don't release this as a signed Maven Repository artifact, nor do we follow that automated release process.

File: src/test/scala/akka/stream/alpakka/xlsx/RowTest.scala [hunk 6]
```diff
@@ -49,6 +50,7 @@ final class RowTest extends WordSpec with Matchers with GeneratorDrivenPropertyC
         }
       }
 
+      @nowarn //up-to-date scalacheck uses deprecated Stream instead of LazyList for now
```
Comment (ddworak) [inline src/test/scala/akka/stream/alpakka/xlsx/RowTest.scala:53]: nit: could be more scoped (`Stream.empty: @nowarn`)

File: version.sbt [hunk 7]
```diff
@@ -1 +1 @@
-version in ThisBuild := "0.0.8-AV4"
+version in ThisBuild := "0.0.9-AV4-test"
```
Comment (ddworak) [inline version.sbt:1]: Would you be so kind to also update `.gitignore` for IntelliJ usage? It's pretty annoying rn.

**Review by ddworak — CHANGES_REQUESTED** [review]
I think we should consider merging upstream as well - they've also upgraded to 2.13 (even if a long time ago), there are some non-trivial changes they made then.

File: project/build.properties [hunk 8]
```diff
@@ -1 +1 @@
-sbt.version=1.1.5
+sbt.version=1.4.5
```
Comment (ddworak) [inline project/build.properties:1]: There's actually `1.4.6` available for a while now.

**Comment by mowczare** [comment]
> I think we should consider merging upstream as well - they've also upgraded to 2.13 (even if a long time ago), there are some non-trivial changes they made then.

Nothing cool really. There are 3 commits since fork, 2 of them are just version bumping, and this is the only commit actually doing sth - it doesn't introduce anything but scala 2.13:
https://github.com/schmitch/akka-xlsx/commit/7589892251f25aab8c4a85aa96b37180446e5666

File: build.sbt (hunk 4 above)
Comment (mowczare) [reply build.sbt:10]: That's just my laziness, but you're right, updated scalatest to newest version.

File: build.sbt (hunk 1 above)
Comment (mowczare) [reply build.sbt]: What I can say... Yesterday I had really good reason (version conflict indication). Today I cannot find it :)

File: .idea/codeStyles/Project.xml [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+      </option>
+    </JavaCodeStyleSettings>
+    <ScalaCodeStyleSettings>
+      <option name="alwaysUsedImports">
+        <array>
+          <option value="io.udash._" />
+        </array>
+      </option>
+      <option name="PLACE_SELF_TYPE_ON_NEW_LINE" value="false" />
+      <option name="BLANK_LINES_AROUND_METHOD_IN_INNER_SCOPES" value="0" />
```
Comment (ddworak) [inline .idea/codeStyles/Project.xml:39]: I think `<option name="FORMATTER" value="1" />` should be added due to scalafmt usage.

File: project/plugins.sbt [hunk 10]
```diff
@@ -1,8 +1,4 @@
 logLevel := Level.Warn
 
 // Dependency Resolution
-updateOptions := updateOptions.value.withGigahorse(false)
-
-// Publishing
-addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.1.0")
-addSbtPlugin("com.github.gseitz" % "sbt-release" % "1.0.7")
\ No newline at end of file
+updateOptions := updateOptions.value.withGigahorse(false)
```
Comment (ddworak) [inline project/plugins.sbt:4]: nit: not needed with our publishing

File: version.sbt (hunk 7 above)
Comment (ddworak) [inline version.sbt:1]: Remember to update the version later on

**Review by ddworak — APPROVED** [review]
LGTM, please verify with our use-case and handle minor comments that are left.

**Pull request merged**